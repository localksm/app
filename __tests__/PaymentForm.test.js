import 'react-native';
import React from 'react';
import { act } from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import {
  PaymentForm,
  PaymentFormInputs,
  FormTextInp,
  FormTextAreaInp,
  FormInput,
} from '../src/molecules';

configure({ adapter: new Adapter() });

it('renders correctly', async () => {
  jest.useFakeTimers();
  await act(() => {
    const onChange = (value) => jest.fn(value);
    const wrapper = shallow(
      <PaymentForm show method="VE" onChangeText={onChange} errors={{}} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});

it('renders payment text input', async () => {
  jest.useFakeTimers();
  await act(() => {
    const handleChangeMock = jest.fn();
    const wrapper = shallow(
      <FormTextInp
        element="name"
        placeholder="name"
        keyboardType="default"
        autoCapitalize="sentences"
        onChangeText={handleChangeMock}
        errors={{}}
      />,
    );
    wrapper.find('InputText').simulate('changeText', 'test text');
    expect(handleChangeMock).toHaveBeenCalledWith('name', 'test text');
    expect(wrapper).toMatchSnapshot();
  });
});

it('renders payment text textarea', async () => {
  jest.useFakeTimers();
  await act(() => {
    const handleChangeMock = jest.fn();
    const wrapper = shallow(
      <FormTextAreaInp
        element="name"
        placeholder="name"
        onChangeText={handleChangeMock}
        errors={{}}
        styles={{}}
      />,
    );
    wrapper.find('Textarea').simulate('changeText', 'test text');
    expect(handleChangeMock).toHaveBeenCalledWith('name', 'test text');
    expect(wrapper).toMatchSnapshot();
  });
});

it('renders payment input type text input', async () => {
  jest.useFakeTimers();
  await act(() => {
    const handleChangeMock = jest.fn();
    const wrapper = shallow(
      <FormInput
        element="name"
        placeholder="name"
        keyboardType="default"
        autoCapitalize="sentences"
        onChangeText={handleChangeMock}
        errors={{}}
        styles={{}}
        type="input"
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});


it('renders payment input type text textarea', async () => {
  jest.useFakeTimers();
  await act(() => {
    const handleChangeMock = jest.fn();
    const wrapper = shallow(
      <FormInput
        element="name"
        placeholder="name"
        keyboardType="default"
        autoCapitalize="sentences"
        onChangeText={handleChangeMock}
        errors={{}}
        styles={{}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});

it('renders paymentInputs correctly', async () => {
  jest.useFakeTimers();
  await act(() => {
    const onChange = (value) => jest.fn(value);
    const wrapper = shallow(
      <PaymentFormInputs show method="VE" onChangeText={onChange} errors={{}} styles={{}} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});