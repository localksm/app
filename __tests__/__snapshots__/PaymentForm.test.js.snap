// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `ShallowWrapper {}`;

exports[`renders payment input type text input 1`] = `ShallowWrapper {}`;

exports[`renders payment input type text textarea 1`] = `ShallowWrapper {}`;

exports[`renders payment text input 1`] = `ShallowWrapper {}`;

exports[`renders payment text textarea 1`] = `ShallowWrapper {}`;

exports[`renders paymentInputs correctly 1`] = `ShallowWrapper {}`;
